src/NServiceBus.AcceptanceTests/Routing/When_publishing_pre_created_interface.cs

```diff
@@ -0,0 +1,115 @@
+﻿namespace NServiceBus.AcceptanceTests.Routing
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using Features;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+    using Conventions = AcceptanceTesting.Customization.Conventions;
+
+    public class When_publishing_pre_created_interface : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_publish_event_to_interface_type_subscribers()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Publisher>(b =>
+                    b.When(c => c.Subscribed, (session, ctx) => session.SendLocal(new StartMessage())))
+                .WithEndpoint<Subscriber>(b => b.When(async (session, ctx) =>
+                {
+                    await session.Subscribe<MyEvent>();
+                    if (ctx.HasNativePubSubSupport)
+                    {
+                        ctx.Subscribed = true;
+                    }
+                }))
+                .Done(c => c.GotTheEvent)
+                .Run();
+
+            Assert.True(context.GotTheEvent);
+            Assert.AreEqual(typeof(MyEvent), context.EventTypePassedToRouting);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool GotTheEvent { get; set; }
+            public bool Subscribed { get; set; }
+            public Type EventTypePassedToRouting { get; set; }
+        }
+
+        public class Publisher : EndpointConfigurationBuilder
+        {
+            public Publisher()
+            {
+                EndpointSetup<DefaultPublisher>(c =>
+                {
+                    c.Pipeline.Register("EventTypeSpy", new EventTypeSpy((Context)ScenarioContext), "EventTypeSpy");
+                    c.OnEndpointSubscribed<Context>((s, context) =>
+                    {
+                        if (s.SubscriberEndpoint.Contains(Conventions.EndpointNamingConvention(typeof(Subscriber))))
```

Comment:
Looks like something changed here between 7.x and 6.x. `SubscriberEndpoint` doesn't exist.